cmd/download.go

```diff
@@ -47,9 +55,22 @@ func runFetch(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+
+	// Ensure ARTIFACTORY_TOKEN is set if cache URLs require it (new part starts here)
+	token := os.Getenv("ARTIFACTORY_TOKEN")
+	if token == "" {
+		log.Warn().Msg("ARTIFACTORY_TOKEN is not set; any cache URLs requiring authentication may fail.")
+	} else {
+		log.Debug().Msg("ARTIFACTORY_TOKEN is set; proceeding with potential cache authentication.")
+	}
+	// New part ends here
+
+	// Proceed with the downloading logic
 	err = lock.Download(dir, tags, notags, perm)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to download resources: %w", err)
```

Comment:
Same comment as mentioned previously: is the error message from the failed function insufficient?  Should it be addressed there?